### ISSUE TITLE
TorchScript Model Saved on x86 Returns NaNs When Loaded on s390x

### ISSUE THREAD

**Issue by KanakaPathivada** [pytorch/pytorch#151428]
### 🐛 Describe the bug

I trained a simple LSTM model on the Iris dataset using PyTorch on an **x86 (little-endian)** system. I then saved the model using both `torch.jit.trace()` and `torch.jit.script()`.

When loading this `.pt` scripted model on an **s390x (big-endian)** architecture, the model loads **without error**, but the output is **entirely NaN**, even when using `set_default_load_endianness(LoadEndianness.LITTLE)` before loading.

This issue does **not occur** when loading the model on x86 again. Only the s390x platform exhibits this behavior.

**Reproduction Script**

- Save the model on x86_64
```
import torch
import torch.nn as nn
import torch.nn.functional as F
from sklearn.datasets import load_iris
from sklearn.preprocessing import StandardScaler
# Set deterministic behavior
torch.manual_seed(0)
# Define a simple LSTM model
class LSTMClassifier(nn.Module):
    def __init__(self, input_size, hidden_size, num_classes):
        super(LSTMClassifier, self).__init__()
        self.lstm = nn.LSTM(input_size, hidden_size, batch_first=True)
        self.fc = nn.Linear(hidden_size, num_classes)
def forward(self, x):
        x = x.unsqueeze(1)  # (batch, seq_len=1, input_size)
        lstm_out, _ = self.lstm(x)
        return self.fc(lstm_out[:, -1, :])  # logits
# Load data
X = load_iris().data
X = StandardScaler().fit_transform(X)
X_tensor = torch.tensor(X, dtype=torch.float32)
# Train dummy model
model = LSTMClassifier(input_size=4, hidden_size=16, num_classes=3)
criterion = nn.CrossEntropyLoss()
optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
y = torch.randint(0, 3, (150,))
for epoch in range(10):
    out = model(X_tensor)
    loss = criterion(out, y)
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()
# Save with torch.jit.script
scripted_model = torch.jit.script(model)
torch.jit.save(scripted_model, "lstm_script_model.pt")
# Save with torch.jit.trace
traced_model = torch.jit.trace(model, torch.randn(1, 4))
torch.jit.save(traced_model, "lstm_traced_model.pt")
```

- Load the model on s390x (zLinux)
```
import torch
from torch.serialization import set_default_load_endianness
from torch.utils.serialization.config import LoadEndianness
# Set little-endian since model was saved on x86
set_default_load_endianness(LoadEndianness.LITTLE)
# Load the TorchScript model
model = torch.jit.load("lstm_script_model.pt")
model.eval()
# Test input
input_data = torch.tensor([
    [0.1, -0.2, 0.3, 0.4],
    [-1.1, 0.9, -0.3, 0.7],
    [1.2, -1.2, 1.3, -1.1],
    [0.3, 0.5, -0.4, 0.2]
], dtype=torch.float32)
# Inference
with torch.no_grad():
    output = model(input_data)
print("Output:")
print(output)

# Load the TorchScript trace model
model = torch.jit.load("lstm_trace_model.pt")
model.eval()
# Inference
with torch.no_grad():
    output = model(input_data)
print("Output:")
print(output)
```

**Observed Behavior**
```
tensor([[nan, nan, nan],
        [nan, nan, nan],
        [nan, nan, nan],
        [nan, nan, nan]])
```
Even though the model loads successfully and is in eval() mode, the inference result is NaN across all outputs.

**Expected Behavior**
The model should return consistent logits as it does on x86.
```
tensor([[-4.2518,  6.0927, -1.8566],
        [ 6.4866, -2.5016, -3.5421],
        [-3.9846, -3.5373,  6.0734],
        [-4.0324,  4.9734, -1.0101]])
```

**Request**
Would appreciate help in:
- Is this a known TorchScript limitation across different architectures (x86 vs s390x)?
- Are there any workarounds (e.g., manually loading LSTM weights)?
- Could this be a bug in how TorchScript handles LSTM/any other model's weight serialization across architectures?

### Versions

```
PyTorch version: 2.5.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (s390x)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.10.12 (main, Feb  4 2025, 14:57:36) [GCC 11.4.0] (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

CPU:
Architecture:                         s390x
CPU op-mode(s):                       32-bit, 64-bit
Byte Order:                           Big Endian

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] torch==2.5.0
```

**Comment by AlekseiNikiforovIBM** [comment]
I'm working on it.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
